tmt/base.py

```diff
@@ -528,11 +524,40 @@ def environment(self):
         """ Return combined environment from plan data and command line """
         if self.my_run and self.my_run.environment:
             combined = self._environment.copy()
+
+            # vars sets in cli takes precedence
             combined.update(self.my_run.environment)
             return combined
         else:
             return self._environment
 
+    @staticmethod
+    def _get_environment_vars(node):
+        # Environment variables from files
+        env_files = node.get("environment-file") or []
+        if not isinstance(env_files, list):
+            raise tmt.utils.GeneralError(
+                f"environment-file parameter should be a list. "
+                f"Received {type(env_files)}"
+                )
+        env_files_vars = tmt.utils.environment_file_to_dict(env_files)
+        environment = env_files_vars
+
+        # Environment variables, make sure that values are string
+        env_vars = dict([
+            (key, str(value)) for key, value
+            in node.get('environment', dict()).items()])
+
+        # combine all env sources into one (env_vars takes precendence)
+        environment.update(**env_vars)
+
+        # ensure that presented vars in os.environment was not overwritten
+        # and revert these values if present
+        for var in (
+                env_var for env_var in environment if env_var in os.environ):
+            environment[var] = os.environ[var]
+        return environment
```

Comment:
@thrix, @lukaszachy, @FrNecas, is that right that the variable from the process should have the highest priority?

    VAR=shell tmt run --environment VAR=option

So for the example above `shell` value should be the winner? Just want to ensure we have a clear agreement on this...

Reply:
I do not even agree we should by default forward vars from tmt's environment :) If needed, should be another special option ...

Reply:
taking as example podman: It has a special option `    --env-host                                 Use all current host environment variables in container`

Reply:
and I believe the highest priority should have the option passed to `--environment`

Reply:
This is not forwarding the whole `tmt` process environment, only for cases when the same variable is defined in both `environment` key and process environment.

Reply:
I do not see a reason why we would complicate it like this, the user can in that case just use:

```
VAR=shell
tmt run --environment VAR=$VAR
```

Reply:
I believe that proposal is not obvious, is that something common which I do not know?

Reply:
The special option, as done in `podman`, make sense to me.

Reply:
I believe the feature is dangerous from security perspective. Let's say I have some secrets in my env var, like AWS_SECRET_ID, somebody could then read this variable from a tmt test if he sets env var `AWS_SECRET_ID` ?

Reply:
Agreed, let's keep this functionality for a dedicated option, if really needed.

Reply:
ok, then, I just remove this part code, which restore the vars to the os.environ. @thrix thx for spoting this

Reply:
resolved